### PR TITLE
Replace "unix" by "http+unix" for PROTOCOL

### LINKED ISSUE #9318: Rename existing unix protocol to http+unix
With #9298 there is now an `fcgi+unix` protocol which allows serving FastCGI via a unix socket.

Personally, I feel that it makes sense to rename the existing `unix` to `http+unix`. We can still accept `unix`, just warning that it's defaulting to `http+unix` and that you should probably use `http+unix` instead.

### PR DESCRIPTION
**Pull request by mscherer**
Fixes #9318

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -597,16 +598,16 @@ func NewContext() {
...
-		UnixSocketPermissionRaw := sec.Key("UNIX_SOCKET_PERMISSION").MustString("666")
-		UnixSocketPermissionParsed, err := strconv.ParseUint(UnixSocketPermissionRaw, 8, 32)
-		if err != nil || UnixSocketPermissionParsed > 0777 {
-			log.Fatal("Failed to parse unixSocketPermission: %s", UnixSocketPermissionRaw)
+	case "fcgi+unix", "unix", "http+unix":
+		switch protocolCfg {
+		case "fcgi+unix":
+			Protocol = FCGIUnix
+		case "unix":
+			Protocol = HTTPUnix
```
Comment (techknowlogick) [inline modules/setting/setting.go]: A very tiny nitpick: You could use fallthrough after the log instead of this line.

Comment (mscherer) [reply modules/setting/setting.go]: Oh, you're right, that's better. I am still a noob in Go, so I didn't knew that was a option.

**Review by zeripath — DISMISSED** [review]
Edit I misread the pr I see that "unix" is still supported.

**Review by silverwind — APPROVED** [review]

**Comment by zeripath** [comment]
Is this really worth doing?

I'd prefer if we could get host and port merged in to an ADDRESS value and then we'd be able to detect if we need to use a unix or a tcp port by the ADDRESS. Then protocol can drop all of the unix variants.

File: modules/setting/setting.go (hunk 1 above)
Comment (techknowlogick) [reply modules/setting/setting.go]: No worries:) Glad you were able to learn a new thing about go. Maybe more PRs are in your future ;)

**Comment by mscherer** [comment]
I may be missing something, but a ADDRESS value wouldn't be sufficient, as you still need to specify if that's FCGI or HTTP (or a future protocol), no ?

**Comment by zeripath** [comment]
Yes you would still need the protocol field but the weird +unix variants would be unnecessary

**Comment by mscherer** [comment]
Yeah, I agree, this would be better. having each potential protocol being double is not elegant and would make doc more complex quite fast if a new protocol is added (eg, something to replace FastCGI).

**Comment by silverwind** [comment]
Keep note that`https+unix` is certainly also a possibility.

File: modules/setting/setting.go [hunk 2]
```diff
@@ -597,16 +598,16 @@ func NewContext() {
 		}
 	case "fcgi":
 		Protocol = FCGI
-	case "fcgi+unix":
-		Protocol = FCGIUnix
-		UnixSocketPermissionRaw := sec.Key("UNIX_SOCKET_PERMISSION").MustString("666")
```
Comment (lunny) [inline modules/setting/setting.go:625]: Why we removed these logic?

Comment (techknowlogick) [reply modules/setting/setting.go:625]: I believe this just removes the duplication, the logic still exists below

Comment (lunny) [reply modules/setting/setting.go:625]: I don't think so. Following lines checked `700` but new code lost that.

File: modules/setting/setting.go [hunk 3]
```diff
@@ -597,16 +598,16 @@ func NewContext() {
...
+			fallthrough
+		case "http+unix":
+			Protocol = HTTPUnix
 		}
-		UnixSocketPermission = uint32(UnixSocketPermissionParsed)
-	case "unix":
-		Protocol = UnixSocket
 		UnixSocketPermissionRaw := sec.Key("UNIX_SOCKET_PERMISSION").MustString("666")
 		UnixSocketPermissionParsed, err := strconv.ParseUint(UnixSocketPermissionRaw, 8, 32)
 		if err != nil || UnixSocketPermissionParsed > 0777 {
```
Comment (techknowlogick) [inline modules/setting/setting.go:638]: @lunny that 777 check is here

File: modules/setting/setting.go (hunk 2 above)
Comment (mscherer) [reply modules/setting/setting.go:625]: I refactored the code, but indeed the diff do not make it apparent. The case "unix" and "fcgi+unix" was the same, just a different Protocol value. So I merged them, and added the switch/case for the protocol and warning.

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
This change is fine to me.

zeripath's suggestion about "get host and port merged in to an ADDRESS" is pretty good.

So should we wait for the new work about unified ADDRESS, or get this PR merged first?

(I marked this PR as `wip` temporarily)

**Comment by mscherer** [comment]
I feel that the ADDRESS change would requires a longer design discussion.

**Comment by mscherer** [comment]
I rebased, but I think it requires a quick re-review

**Comment by wxiaoguang** [comment]
@zeripath @techknowlogick how do you think?

**Comment by wxiaoguang** [comment]
Bump. If still no objection in 2 days, maybe we can get this PR merged.

**Comment by zeripath** [comment]
yeah I think we should merge as is and address the address issue at another point.

**Comment by lunny** [comment]
> Bump. If still no objection in 2 days, maybe we can get this PR merged.

So do you want to remove the `wip` label?

**Pull request merged**